client/python/openlineage/client/transport/file.py

```diff
@@ -47,8 +40,13 @@ def emit(self, event: RunEvent | DatasetEvent | JobEvent) -> None:
         if self.config.append:
             log_file_path = self.config.log_file_path
         else:
-            timestr = datetime.now().strftime("%Y%m%d-%H%M%S.%f")
-            log_file_path = f"{self.config.log_file_path}-{timestr}"
-
-        with open(log_file_path, "a" if self.config.append else "w") as log_file_handle:
-            log_file_handle.write(Serde.to_json(event) + "\n")
+            time_str = datetime.now().strftime("%Y%m%d-%H%M%S.%f")
+            log_file_path = f"{self.config.log_file_path}-{time_str}"
+
+        try:
+            with open(log_file_path, "a" if self.config.append else "w") as log_file_handle:
+                log_file_handle.write(Serde.to_json(event) + "\n")
+        except (PermissionError, io.UnsupportedOperation) as error:
```

Comment:
Are those two coming from internal implementation of `open` built-in?

Reply:
When there is no write permission to the file, the `PermissionError` is raised when calling `open`.  If we changed the open mode to `r`, we would get `io.UnsupportedOperation` when calling `.write()`. Also, both error messages do not include file path so i added that in the custom error message.